online-inference/hf-llm/service/service.py

```diff
@@ -135,8 +144,10 @@ def load_model(path_uri: str) -> TextGenerationPipeline:
     logger.info(f"Loading {args.precision} model from: {path_uri}")
     if path_uri.startswith("s3://"):
         return load_model_s3(path_uri)
-    else:
+    elif os.path.exists(f"{path_uri}/model.tensors"):
         return load_model_local(path_uri)
```

Comment:
The `load_model_local` case loading a local tensorized model does not cast to or check for the requested precision. (The S3 case handles this by automatically choosing the correct path in the `tensorized` bucket between `.../model.tensors` or `.../fp16/model.tensors`, but for local model loading, that path structure being available is less likely.)